src/signalHandler.cpp

```diff
@@ -0,0 +1,55 @@
+#include "signalHandler.hpp"
+
+// NOLINTNEXTLINE: This is a global variable used in signalHandler
+volatile std::sig_atomic_t g_signalStatus = 0;
+
+/**
+ * @brief Simple signal handler that sets the global variable g_signalStatus to the signal number.
+ *
+ * @param signal The signal number.
+ */
+void signalHandler(int signal) { g_signalStatus = signal; }
+
+/**
+ * @brief Converts a signal number to its name.
+ *
+ * @param signum The signal number.
+ * @return std::string The signal name.
+ */
+std::string signalNumToName(int signum)
```

Comment:
No, on my machine I have 64. 
I picked the first 15 since SIGTERM (which is 15) is "typically used to administratively terminate a process". If its another signal the function returns "Unkown signal"

I don't think we need more: we control which signal we catch (by registering the signal handler) and subsequently use as a shutdown signal.

Right now in main only SIGINT is registered, which means any other signal would lead to sudden shutdown (if the default behavior is not ignore, like with SIGCHLD)

Reply:
So we only process SIGTERM, but are able to resolve the name of 14 other signals just so we know what signal was sent?

Reply:
Well, this is just a sigNum to sigName helper function.
If we receive a signal number and just want to print it out, we can.

Reply:
Yes, that's what I meant with resolving. But will we ever print out any of the other signals?

Reply:
When the program is running perfectly fine: no. I don't think so. 
When debugging? Well depends on the signal catched, aka how the program evolves. Do we use another signal? Maybe even someplace else?

I could also hard code the string of the signal name it if that seems like a better option?

I have to admit: this discussion seems very [bike-sheddy](https://en.wikipedia.org/wiki/Law_of_triviality) to me. Why do we care how many signals we can print out or not? The much more interesting discussion for me is what signals do we want to catch as a way of shutting down the server? Do we want different behavior depending on signal received?

Reply:
Suggestion: I just move the function to LogOstreamInserters.cpp, which has other debug printers?

Reply:
The purpose of the question was to induce reflection on whether only SIGTERM is relevant - which it did. It is insofar non-trivial precisely because it touches the topic of what signals are actually required to be implemented, and moreover which are not. 

Here are some sources on this subject which state that at least SIGHUP and SIGQUIT should also be handled:

- https://nginx.org/en/docs/control.html

- https://rafallorenz.com/go/handle-signals-to-graceful-shutdown-http-server/

>  Why do we care how many signals we can print out or not?
Because it is good practice to only implement code which is actually used and [avoid dead code](https://stackoverflow.com/questions/15699995/could-someone-explain-the-pros-of-deleting-or-keeping-unused-code). Especially in such an obvious case.

Reply:
Ah, right.

I suggest if a general topic arises from a reflection inducing non-trivial question to break it out into a new comment. 

> avoid dead code

I don't really understand how this simple conversion function generates confusion, introduces bugs and maintenance burden. Alas here we are.
Removed all conversions except for SIGHUP, SIGINT, SIGQUIT, SIGTERM